maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultMavenPluginManager.java

```diff
@@ -303,25 +308,36 @@ public MojoDescriptor getMojoDescriptor( Plugin plugin, String goal, List<Remote
         return mojoDescriptor;
     }
 
-    public void checkRequiredMavenVersion( PluginDescriptor pluginDescriptor )
+    
+    @Override
+    public void checkPrerequisites( PluginDescriptor pluginDescriptor )
         throws PluginIncompatibleException
     {
-        String requiredMavenVersion = pluginDescriptor.getRequiredMavenVersion();
-        if ( StringUtils.isNotBlank( requiredMavenVersion ) )
-        {
-            try
-            {
-                if ( !runtimeInformation.isMavenVersion( requiredMavenVersion ) )
-                {
-                    throw new PluginIncompatibleException( pluginDescriptor.getPlugin(),
-                                                           "The plugin " + pluginDescriptor.getId()
-                                                               + " requires Maven version " + requiredMavenVersion );
-                }
-            }
-            catch ( RuntimeException e )
-            {
-                logger.warn( "Could not verify plugin's Maven prerequisite: " + e.getMessage() );
-            }
+        List<IllegalStateException> prerequisiteExceptions = new ArrayList<>();
+        prerequisiteCheckers.forEach( c -> 
+        {
+            try 
+            { 
+                c.accept( pluginDescriptor );
+            } 
+            catch ( IllegalStateException e )
+            {
+                prerequisiteExceptions.add( e );
+            }
+        } );
+        // aggregate all exceptions
+        if ( !prerequisiteExceptions.isEmpty() )
+        {
+            String messages = prerequisiteExceptions.stream()
+                            .map( IllegalStateException::getMessage )
+                            .collect( Collectors.joining( ", " ) );
+            PluginIncompatibleException pie  = new PluginIncompatibleException( pluginDescriptor.getPlugin(),
+                                                   "The plugin " + pluginDescriptor.getId()
+                                                       + " has unmet requirements: " 
```

Comment:
requirements >= prerequisites?

Reply:
done in https://github.com/apache/maven/pull/827/commits/04cffe8f70ae94111e84ed44eaa810aec6a69541.